### ISSUE TITLE
Docs changes

### ISSUE THREAD

**Issue by Aziz-repo** [pylessard/python-udsoncan#151]
Hi,
This period I'm working on a project involving a ECU in an automotive industry and I frequently check the docs presented in [https://udsoncan.readthedocs.io/en/stable/index.html](url) .
I checked the docs today and I noticed that some of the docs was not to be found. (Particularly in the services and client sections), I checked both versions (stable and latest). 
I noticed also that a merge request was accepted and the master branch was updated but even if when I checked the changes the docs where not changed.

**Comment by pylessard** [comment]
You are right, something is broken on readthedocs.

Let me see

**Comment by pylessard** [comment]
Ok. So I released V1.17 yesterday.
The big change in 1.17 is that I introduced type hints.  I used the `TypedDict` class at few places that is not supported by python 3.7 and readthedocs.org uses python 3.7.   The build log was full of ImportError.

I added a stub for TypedDict for python versions below 3.8. It fixes the docs and also fixes the backward compatibility issue that came iwth TypeDict (I think). 

Will release 1.17.1 with that fix. 
Thank you
